src/activity-logger/background/log-page-visit.js

```diff
@@ -52,17 +58,23 @@ export async function logPageVisit({ tabId, url }) {
     // The time to put in documents.
     const timestamp = Date.now()
 
-    // First create an identifier for the page being visited.
-    const storePageResult = await storePage({ tabId, url })
+    const pageId = generatePageDocId({ url })
+    const existingPage = await index.initSingleLookup()(pageId)
+
+    let storePageResult
+    if (existingPage == null || timestamp - existingPage.latest > threshold) {
```

Comment:
Would be really good to have a plain-English explanation for complex conditions like this, as it's not that obvious at first glance. Can do as a comment or, even better, in the JSDoc for this function.